src/actions/editor.js

```diff
@@ -323,24 +295,32 @@ const executeSendRequest = (formValues, contentLanguages, user, updateExisting,
                 }
 
                 if(response.status === 200 || response.status === 201) {
-                    // create sub events after creaing super event successfully
+                    // create or update sub events after creating/updating super event successfully
                     if (json.super_event_type === 'recurring') {
-                        const formWithAllSubEvents = combineSubEventsFromEditor(formValues, updateExisting)
-                        dispatch(
-                            sendRecurringData(formWithAllSubEvents, contentLanguages, user, updateExisting, publicationStatus, json['@id'])
-                        )
-                        dispatch(sendDataComplete(json, actionName))
+                        // update sub events
+                        if (updateExisting) {
+                            const subEventsToUpdate = getState().subEvents.items
```

Comment:
Hm. Looks like the function `combineSubEventsFromEditor` does this already, if you call it with `updateExisting=True`. It's a bit funny to first check for `updateExisting=False` and only then call `combineSubEventsFromEditor`.